File: 2236c.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    while (t--) {
        int a, b, x;
        cin >> a >> b >> x;
        int ans = 1e18, s = 0;
        while (a != b) {
            ans = min(ans, s + abs(a - b));
            s++;
            if (a > b) a /= x;
            else b /= x;
        }
        cout << min(ans, s) << '\n';
    }

    return 0;
}

File: 2026/2236/2236c.py
t = int(input())
for _ in range(t):
    a, b, x = map(int, input().split())
    A = [a]
    B = [b]
    while a > 0:
        a //= x
        A.append(a)
    while b > 0:
        b //= x
        B.append(b)
    ans = 10 ** 18
    for i in range(len(A)):
        for j in range(len(B)):
            ans = min(ans, abs(A[i] - B[j]) + i + j)
    print(ans)
